SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -662,21 +671,67 @@ private int numInstancesExpected(SingularityRequest request, SingularityPendingR
         LOG.info("Scheduling requested immediate run of {}", request.getId());
       } else {
         try {
-          final CronExpression cronExpression = new CronExpression(request.getQuartzScheduleSafe());
-
+          long nextRunAtTimestamp = 0;
+          Date nextRunAtDate = null;
           final Date scheduleFrom = new Date(now);
-          final Date nextRunAtDate = cronExpression.getNextValidTimeAfter(scheduleFrom);
-
-          if (nextRunAtDate == null) {
-            return Optional.absent();
+          if(request.getScheduleTypeSafe() == ScheduleType.RFC5545)
+          {
+            final String schedule = request.getSchedule().get();
+            final RecurrenceRule recurrenceRule = new RecurrenceRule(schedule);
+            if(recurrenceRule.isInfinite())
+            {
+              // set limit at 2100-01-01 00:00:00
+              recurrenceRule.setUntil(new DateTime(2100, 1, 1, 0, 0, 0));
+            }
+
+            // DTSTART is RFC5545 but NOT in the recur string, but its a nice to have
+            Pattern pattern = Pattern.compile("DTSTART=([0-9]{8}T[0-9]{6})");
+            Matcher matcher = pattern.matcher(schedule);
+            DateTime startDateTime;
+            org.joda.time.DateTime dtStart;
+            if(matcher.find())
+            {
+              DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyyMMdd'T'HHmmss");
+              dtStart = formatter.parseDateTime(matcher.group(1));
+            }
+            else{
+              dtStart = org.joda.time.DateTime.now();
+              dtStart = dtStart.withSecondOfMinute(0);
+            }
+
+            startDateTime = new DateTime(dtStart.getYear(),(dtStart.getMonthOfYear() - 1), dtStart.getDayOfMonth(),
+                    dtStart.getHourOfDay(), dtStart.getMinuteOfHour(), dtStart.getSecondOfMinute());
+            RecurrenceRuleIterator timeIterator = recurrenceRule.iterator(startDateTime);
```

Comment:
I think we can't do that, because `DTSTART` is used to loop by the library. So if DTSTART is a date in the past, `org.dmfs.rfc5545.recur.RecurrenceRule` will start looping from that day to get the next best date & time that is in the future.
I'm almost done refactoring this into a class, committing it in a bit.